sql/cache.go

```diff
@@ -0,0 +1,131 @@
+package sql
+
+import (
+	"fmt"
+	"hash/crc64"
+	"runtime"
+
+	lru "github.com/hashicorp/golang-lru"
+	errors "gopkg.in/src-d/go-errors.v1"
+)
+
+var table = crc64.MakeTable(crc64.ISO)
+
+// CacheKey returns a hash of the given value to be used as key in
+// a cache.
+func CacheKey(v interface{}) uint64 {
+	return crc64.Checksum([]byte(fmt.Sprintf("%#v", v)), table)
+}
+
+// ErrKeyNotFound is returned when the key could not be found in the cache.
+var ErrKeyNotFound = errors.NewKind("memory: key %d not found in cache")
+
+type lruCache struct {
+	memory Freeable
+	r      Reporter
+	size   int
+	cache  *lru.Cache
+}
+
+func newLRUCache(memory Freeable, r Reporter, size int) *lruCache {
+	lru, _ := lru.New(size)
+	return &lruCache{memory, r, size, lru}
+}
+
+func (l *lruCache) Put(k uint64, v interface{}) error {
```

Comment:
I saw  in other parts of the codebase that the functions implementing an interface had documentation as in "This implements KeyValue interface". I don't know if that's some kind of convention we have (is it?), but it definitely helps me when I am reading new code, so maybe it is worth it to write it here in `Put` and `Get`, as well as in `Free` and `Dispose` below.

Reply:
Adding that doc to methods is basically so the linter doesn't complain, because it doesn't really add much value. The reason this is not commented is because the type is private, and thus the linter does not complain about it.